bin/build_latest.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# Helpers for checking out and building latest version of Istio (with minimal/no use of
+# manual SHAs). This is an in-progress proposal/PoC for a consistent build.
+
+# Environment:
+# ISTIO_BASE: base directory where istio will be checked out or built. Defaults to GOPATH.
+#  The script will populate or update the version.
+# HUB: hub to use to upload the docker images.
+# TAG: tag to use for the docker images. Defaults to user ID.
+
+HUB=${HUB:-gcr.io/istio-testing}
+ISTIO_IO=${ISTIO_BASE:-${GOPATH:-$HOME/go}}/src/istio.io
+
+
+# Build all components using bazel.
+function istio_build() {
+
+  # Note: components may still use old SHA - but the test will build the binaries from master
+  # from each component, to make sure we don't test old code.
+  pushd $ISTIO_IO/pilot
+  bazel build ...
+  ./bin/init.sh
+  popd
+
+  (cd $ISTIO_IO/mixer; bazel build ...)
+
+  (cd $ISTIO_IO/proxy; bazel build tools/deb/... src/envoy/mixer:envoy)
+
+  (cd $ISTIO_IO/auth; bazel build ...)
+}
+
+# Pull from master, equivalent with "repo sync"
+function istio_sync() {
+  # TODO: use "repo sync" instead
+  # TODO: sync on green build ( if repo is used )
+  mkdir -p $ISTIO_IO
+
+  for sub in pilot istio mixer auth proxy; do
+    if [[ -d $ISTIO_IO/$sub ]]; then
+      echo "Syncing $sub"
+      (cd $ISTIO_IO/$sub; git pull origin master)
```

Comment:
if you use GOPATH, this might just screw up your setup.

Reply:
The intent is to behave like 'repo sync' - i.e. update your workspace with latest changes from
master. Once we switch to real repo, it'll be smarter and stick with branches.

Reply:
+1 on not putting `origin master` here (already made that change multiple times in previous PRs)

instead write a "istio_change_branch ..." maybe (or a "istio_apply_to_all" that applies the same command in all repos; command can be `git stash` or `git checkout otherbranch`...)

Reply:
Also don't use `origin`. I use a different name, and `upstream` is common.

Reply:
Added a env variable to customize.